content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -0,0 +1,176 @@
+  previewFeatures = ["referentialIntegrity"]
+}
+```
+
+To change the relation mode, update the `relationMode` field in the `datasource` block.
+
+<Admonition type="warning">
+
+The <inlinecode>relationMode</inlinecode> field was renamed in Prisma version 4.5.0,
+and was previously named <inlinecode>referentialIntegrity</inlinecode>.
+
+</Admonition>
+
+The available options are:
+
+- `foreignKeys`: this handles relations in the database using foreign keys. This is the default option when using all relational database connectors.
+- `prisma`: this emulates relations in the Prisma client. This is the default (and only available) option when using the MongoDB connector. You should also [enable this option](/guides/database/using-prisma-with-planetscale#how-to-enable-emulation-of-referential-integrity) when using the MySQL connector with a PlanetScale database.
+
+<Admonition type="warning">
+
+When using Prisma Migrate, setting the value to <inlinecode>prisma</inlinecode> will remove all previously created foreign keys in the next migration. Setting the value to <inlinecode>foreignKeys</inlinecode> (either directly or by removing the <inlinecode>relationMode</inlinecode> field and using the default option for relational databases) will create foreign keys for all relations in the next migration.
+
+</Admonition>
+
+<!-- TODO: mention preview feature + version info in intro -->
+
+## Handle relations in your relational database with the <inlineCode>foreignKeys</inlineCode> relation mode
+
+The `foreignKeys` relation mode handles relations in your relational database using foreign keys. This is the default option when you use a relational database connector (PostgreSQL, MySQL, SQLite, SQL Server, CockroachDB).
+
+The `foreignKeys` relation mode is not available when you use the MongoDB connector. Some relational databases, [such as PlanetScale](/guides/database/using-prisma-with-planetscale#how-to-emulate-referential-integrity-in-prisma-client), also forbid the use of foreign keys. In these cases, you should instead [emulate relations in Prisma with the `prisma` relation mode](#emulate-relations-in-prisma-with-the-prisma-relation-mode).
+
+### Referential integrity
+
+The `foreignKeys` relation mode enforces referential integrity checks at the database level. This means that when you _create_ or _update_ a record with a relation to another record, the related record needs to exist. If the record does not exist, the database will return an error message.
+
+### Referential actions
+
+When you use the `foreignKeys` relation mode, referential actions are triggered in the database after you _update_ or _delete_ a record with relations to other records. These referential actions maintain referential integrity in the related records by preventing changes that would break referential integrity, propagating changes through to related records or setting the value of fields that reference the updated or deleted record to a null or default value.
```

Comment:
this happens with the update/delete, so it can return an error instead and fail the query:
```suggestion
When you use the `foreignKeys` relation mode, referential actions are triggered in the database when you _update_ or _delete_ a record with relations to other records. These referential actions maintain referential integrity in the related records by preventing changes that would break referential integrity, cascading changes through to related records or setting the value of fields that reference the updated or deleted record to a `null` or default value.
```